uyuni-tools/last_update_export.py

```diff
@@ -0,0 +1,103 @@
+import socket
+import configparser
+import shutil
+
+# Configuration Variables
+BASE_DIR = "/mnt"  # Define base directory where the exports will be.
+OUTPUT_DIR = os.path.join(BASE_DIR, "export/updates")
+LOG_DIR = os.path.join(BASE_DIR, "logs")
+TODAY = datetime.date.today()
+TARGET_DATE = TODAY - datetime.timedelta(days=1)  # Define the number of days back for export, 1 day by default
+RSYNC_USER = "rsyncuser"  # Define rsync user
+RSYNC_GROUP = "users"     # Define rsync group
+
+def setup_directories(base_path, output_path, log_path):
+    if not os.path.exists(base_path):
+        os.makedirs(base_path, exist_ok=True)
+    if not os.path.exists(log_path):
+        os.makedirs(log_path, exist_ok=True)
+    if os.path.exists(output_path):
+        shutil.rmtree(output_path)
+    os.makedirs(output_path, exist_ok=True)
+
+def setup_logging(LOG_DIR, TODAY):
+    log_file_path = os.path.join(LOG_DIR, f"{TODAY}-daily_export.log")
+    return log_file_path
+
+def create_client():
+    config_path = os.path.expanduser('/root/.mgr-sync')
+    config = configparser.ConfigParser()
+    with open(config_path, 'r') as f:
+        config.read_string('[DEFAULT]\n' + f.read())
+    manager_login = config.get('DEFAULT', 'mgrsync.user')
+    manager_password = config.get('DEFAULT', 'mgrsync.password')
+    suma_fqdn = socket.getfqdn()
+    manager_url = f"https://{suma_fqdn}/rpc/api"
+    context = ssl.create_default_context()
+    client = ServerProxy(manager_url, context=context)
+    return client, client.auth.login(manager_login, manager_password)
+
+# Setup Directories and Logging
+setup_directories(BASE_DIR, OUTPUT_DIR, LOG_DIR)
+log_file_path = setup_logging(LOG_DIR, TODAY)
+
+# Create XML-RPC Client
+client, key = create_client()
+
+# Process channels
+channel_list = client.channel.listVendorChannels(key)
+for channel in channel_list:
+    build_date, channel_label = client.channel.software.getChannelLastBuildById(key, channel["id"]).split()[0], channel["label"]
+    build_date = datetime.datetime.strptime(build_date, "%Y-%m-%d").date()
+
+    if build_date in [TODAY, TARGET_DATE]:
+        channel_OUTPUT_DIR = os.path.join(OUTPUT_DIR, channel_label)
+        os.makedirs(channel_OUTPUT_DIR, exist_ok=True)
+        options_dict = {
+            "outputDir": channel_OUTPUT_DIR,
+            "orgLimit": "2",  # Define the default Organization, 2 by default assuming there is only 1, if multiple set this to the one you assign for exporting.
+            "logLevel": "error",  # Set as 'error' by default but change to 'debug' for detailed logging
+            "packagesOnlyAfter": TARGET_DATE.strftime('%Y-%m-%d')
+        }
+        options = ' '.join([f"--{opt}='{val}'" for opt, val in options_dict.items()])
+        command = f"inter-server-sync export --channels='{channel_label}' {options}"
+        with open(log_file_path, "a") as log_file:
+            subprocess.run(command, shell=True, stdout=log_file, stderr=subprocess.STDOUT, check=True)
```

Comment:
In general I would recommend to use `subprocess.run` without `shell=True` (if possible), as it would prevent potential shell injections. as you do already here https://github.com/uyuni-project/contrib/pull/23/files#diff-828d5f5bff6d9517a822e3e503fa62753bbc631f61c1ffccc88c601bf1111ae2R103